src/app/icd/ICDManager.cpp

```diff
@@ -140,6 +140,8 @@ void ICDManager::UpdateOperationState(OperationalState state)
             {
                 ChipLogError(AppServer, "Failed to set Polling Interval: err %" CHIP_ERROR_FORMAT, err.Format());
             }
+
+            InteractionModelEngine::GetInstance()->GetReportScheduler()->OnActiveModeEntered();
```

Comment:
Using the Singleton::GetInstance makes unit testing the ICDManager more complicated. 
I think it would be possible to pass the pointer to the IMEngine when we call the init of the ICDManager in the server init phase.

Might be also better to just give the pointer to the ReportScheduler to the ICDManager?

Reply:
+1. The IM Engine should have a ref to the report scheduler

Reply:
Now passing in a ref to the report scheduler.

Reply:
Might be something that I'm missing but, ReportScheduler and IM engine belong in app, which depends on ICD manager.

Passing either through ICDmanager::Init() will cause a circular dependency no?